src/app/ipv/middleware.js

```diff
@@ -345,27 +331,28 @@ module.exports = {
       const { pageId } = req.params;
       const { context } = req?.session || "";
 
-      // handles page id validation first
       if (!isValidPage(pageId)) {
         res.status(HTTP_STATUS_CODES.NOT_FOUND);
         return res.render(getTemplatePath("errors", "page-not-found"));
       }
 
-      if (!req.session?.ipvSessionId) {
+      if (!req.session?.ipvSessionId && !req.session?.criSessionId) {
```

Comment:
Hmm I don't think `criSessionId` is something we store in the frontend session? in the mobile cross-browser scenario we store the `clientOauthSessionId` so that it can be used to relay the user back to their RP

Reply:
Yes you are right, I typed out the wrong one here